05.ls/lib/ls.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'io/console/size'
+require 'etc'
+
+def main
+  optparse = OptionParser.new
+  options = optparse.getopts(ARGV, 'alr')
+  option = '-'
+  options.each do |option_array|
+    option += option_array[0] if option_array[1]
+  end
+
+  file_directory_name = ARGV[0]
+  current_files = decide_howto_collect_files(option, file_directory_name)
+
+  files_name_length_list = create_filename_length_list(current_files)
+
+  number_rows = files_name_length_list.size / (IO.console_size[1] / (files_name_length_list.max + 6)) + 1
+
+  for_display_array = change_to_ls_list(current_files, number_rows)
+
+  two_d_elements = for_display_array.max_by(&:length).length
+  for_display_array = insert_nil_to_ls_list(for_display_array, two_d_elements).transpose
+
+  display_ls_result(option, current_files, file_directory_name, for_display_array, files_name_length_list)
+end
+
+def decide_howto_collect_files(option, file_directory_name)
+  globe_arg = option&.include?('a') ? File::FNM_DOTMATCH : ''.to_i
+  current_files = Dir.glob('*', globe_arg).sort
+  current_files = Dir.glob(file_directory_name) unless file_directory_name.nil?
+  current_files = current_files.reverse if option&.include? 'r'
+  current_files
```

Comment:
これもまあ細かい話ですが、
```ruby
    option&.include?('r') ? current_files.reverse : current_files
```
とやれば一行でかけますね。
これに関してはちょっとややこしい書き方になってしまうので、どちらで書くかは好みの問題かと思いますが。（むしろ今の方が読み易くていいかもしれませんね。参考までに）

Reply:
あと、`current_files`はローカル変数なので破壊的処理をしてしまっても特に問題ないと思います。なので、もし自分だったらこう書くかなーと思います。見た目の問題でしかないですが。
```suggestion
  current_files.reverse! if option&.include? 'r'
  current_files
```

Reply:
```ruby
option&.include?('r') ? current_files.reverse : current_files
```
この書き方については、目から鱗でした！
他のメソッドもですが、ずっと、最後に変数だけを書くことに、微妙に違和感がありました！
たしかに、オプションがない場合は、`current_files`を返せますね!😃